mits/models.py

```diff
@@ -0,0 +1,179 @@
+
+    @property
+    def salutation(self):
+        return ' and '.join(applicant.first_name for applicant in self.primary_contacts)
+
+    @property
+    def can_save(self):
+        return c.HAS_MITS_ADMIN_ACCESS or (self.is_new and c.BEFORE_MITS_SUBMISSION_DEADLINE or c.BEFORE_MITS_EDITING_DEADLINE)
+
+    @property
+    def completed_hotel_form(self):
+        return any(a.declined_hotel_space or a.requested_room_nights for a in self.applicants)
+
+    @property
+    def steps_completed(self):
+        if not self.games:
+            return 1
+        elif not self.pictures:
+            return 2
+        elif not self.schedule:
+            return 3
+        elif not self.completed_hotel_form:
+            return 4
+        elif not self.submitted:
+            return 5
+        else:
+            return 6
+
+    @property
+    def completion_percentage(self):
+        return 100 * self.steps_completed // c.MITS_APPLICATION_STEPS
+
+
+class MITSApplicant(MagModel):
+    team_id = Column(ForeignKey('mits_team.id'))
+    attendee_id = Column(ForeignKey('attendee.id'), nullable=True)
+    primary_contact = Column(Boolean, default=False)
+    first_name = Column(UnicodeText)
+    last_name = Column(UnicodeText)
+    email = Column(UnicodeText)
+    cellphone = Column(UnicodeText)
+    contact_method = Column(Choice(c.MITS_CONTACT_OPTS), default=c.TEXTING)
+
+    declined_hotel_space = Column(Boolean, default=False)
+    requested_room_nights = Column(MultiChoice(c.MITS_ROOM_NIGHT_OPTS), default='')
+
+    @property
+    def full_name(self):
+        return self.first_name + ' ' + self.last_name
+
+    def has_requested(self, night):
+        return night in self.requested_room_nights_ints
+
+
+class MITSGame(MagModel):
+    team_id = Column(ForeignKey('mits_team.id'))
+    name = Column(UnicodeText)
+    promo_blurb = Column(UnicodeText)
+    description = Column(UnicodeText)
+    genre = Column(UnicodeText)
+    phase = Column(Choice(c.MITS_PHASE_OPTS))
+    min_age = Column(Integer)
+    min_players = Column(Integer, default=2)
+    max_players = Column(Integer, default=4)
+    personally_own = Column(Boolean, default=False)
+    unlicensed = Column(Boolean, default=False)
+    professional = Column(Boolean, default=False)
+
+
+class MITSPicture(MagModel):
+    team_id      = Column(UUID, ForeignKey('mits_team.id'))
+    filename     = Column(UnicodeText)
+    content_type = Column(UnicodeText)
+    extension    = Column(UnicodeText)
+    description  = Column(UnicodeText)
+
+    @property
+    def url(self):
+        return '../mits_applications/view_picture?id={}'.format(self.id)
+
+    @property
+    def filepath(self):
+        return os.path.join(c.MITS_PICTURE_DIR, str(self.id))
+
+
+class MITSTimes(MagModel):
+    team_id = Column(ForeignKey('mits_team.id'))
+    availability = Column(MultiChoice(c.MITS_SCHEDULE_OPTS))
+    multiple_tables = Column(MultiChoice(c.MITS_SCHEDULE_OPTS))
+
+
+@on_startup
+def add_applicant_restriction():
```

Comment:
This is very similar to what we do in the MIVS codebase, but was just different enough that it seemed reasonable to include this as a separate utility.  If we keep doing this kind of thing, it'll probably make sense to refactor it out into a utility that generates these or something.